.github/workflows/golangci-lint.yml

```diff
@@ -53,3 +53,9 @@ jobs:
 
```

Comment:
To match the way `mage llc` invokes it, try `--whole-files` to the args above. The GH action runs the linter using `--new-from-patch` and grabs the PR patch (e.g. https://github.com/elastic/elastic-agent-libs/pull/30.patch).

Reply:
Yep, I'm just trying to see what can be done to match what `mage llc` is doing in order to have annotations.

But maybe I should be thinking about it the other way. Like can we run `mage llc` in a Github Action and have it output the results in the annotation [format](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-notice-message) that Github recognizes. It appears this can be done natively by golangci-lint with `--out-format=github-actions`.

Reply:
As you said, adding `--whole-files` basically matches the command and it does exactly the same just using the patch file instead of the revision. I'd vote for keeping the official Github action and using `--whole-files` with it. If we find something else we can always create our custom Github action but I don't it's worth it at this point.